### DISCUSSION TITLE
Is any way to clean up basetemp directory after execution?

### DISCUSSION THREAD

**Discussion by lena-kru** [pytest-dev/pytest#10380] [Q&A]
I like to use option `--basetemp` and I still can't find the way how to delete this temporal directory.
Is it possible to do that without any crutches?

**Comment by portyu9** [comment]
Hi @lena-kru 

There are now two cases.

### If you use pytest's normal temporary directory

Current pytest supports:

```ini
[pytest]
tmp_path_retention_policy = none
```

which tells pytest not to retain the numbered `tmp_path` session directories.

### If you explicitly use `--basetemp`

For example:

```text
pytest --basetemp=/some/path
```

pytest removes that directory **before the run** if it already exists, but it does not promise to delete the user-specified base directory after the session.

If you specifically need that explicit directory removed afterward, do the cleanup yourself, for example in `conftest.py`:

```python
import shutil

def pytest_sessionfinish(session, exitstatus):
    basetemp = session.config.getoption("basetemp")
    if basetemp:
        shutil.rmtree(basetemp, ignore_errors=True)
```

So today, `tmp_path_retention_policy = none` is the simplest solution when you do not need a fixed `--basetemp`; explicit `--basetemp` cleanup is still your responsibility after the run.

Please mark this answer as accepted if it helped, thank you.
